include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -657,6 +649,26 @@ Value mxfpScaleBf16(RewriterBase &rewriter, Location loc, Value v, Value scale,
 
 } // namespace LLVM
 
+// -----------------------------------------------------------------------
+// Hardware Indices
+// -----------------------------------------------------------------------
+
+// Returns CTA level thread ID.
+Value getThreadId(OpBuilder &rewriter, Location loc);
+
+// Get the lane ID, which is index of the thread within its warp.
+Value getLaneId(OpBuilder &rewriter, Location loc, unsigned threadsPerWarp);
+
+// Get the lane ID and warp ID.
+std::pair<Value, Value> getLaneAndWarpId(OpBuilder &rewriter, Location loc,
+                                         unsigned threadsPerWarp);
+
+// Emit code to compute the (laneId, warpId, blockId) for the current thread.
+std::tuple</*laneId=*/Value, /*warpId=*/Value, /*blockId=*/Value>
+emitHardwareTuple(Location loc, RewriterBase &rewriter,
```

Comment:
How about using a flag to control what ids to obtain and only maintain the `emitHardwareTuple` method? We can also remove `withCTAOffset`. 

Something like

```
enum HardwareIDFlags : unsigned {
  ComputeThreadID = 1 << 0,
  ComputeLaneID   = 1 << 1,
  ComputeWarpID   = 1 << 2,
  ComputeBlockID  = 1 << 3
};
```

It's up to you though, not very important

Reply:
`getThreadId` might still have to be there

Reply:
Yeah I'm not a fan of this trio of methods. What I'll do instead is remove `emitHardwareTuple` altogether since querying the block ID is kind of orthogonal to the lane and warp id (not derived from the thread id).